src/platforms/PolygonId/components/NoDID.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  FormControl,
+  FormLabel,
+  Input,
+  ModalBody,
+  ModalHeader,
+  Spacer,
+} from "@chakra-ui/react"
+import useUser from "components/[guild]/hooks/useUser"
+import Button from "components/common/Button"
+import FormErrorMessage from "components/common/FormErrorMessage"
+import { useSubmitWithSign } from "hooks/useSubmit"
+import { Stack } from "phosphor-react"
+import { useForm, useWatch } from "react-hook-form"
+import { mutate } from "swr"
+import fetcher from "utils/fetcher"
+
+const NoDID = () => {
+  const { id } = useUser()
+  const { isLoading, onSubmit, error } = useSubmitWithSign(
+    () =>
+      fetcher(`${process.env.NEXT_PUBLIC_POLYGONID_API}/v1/polygon-id/connect`, {
+        method: "POST",
+      }),
+    {
+      onSuccess: () => {
+        mutate(`${process.env.NEXT_PUBLIC_POLYGONID_API}/v1/users/${id}/polygon-id`)
```

Comment:
This could be replaced with an optimistic update as I mentioned in my other comment in `MintPolygonIdProofModalContent.tsx`